dec2024/client.js

```diff
@@ -0,0 +1,132 @@
+        ) {
+          id: flow_id
+          flow {
+            slug
+            team {
+              slug
+            }
+            data
+            publishedFlows: published_flows(
+              limit: 1, 
+              order_by: {created_at: desc}
+            ) {
+              id  
+              data
+            }
+            sessions: lowcal_sessions(
+              where: {
+                locked_at: {_is_null: true},
+                submitted_at: {_is_null: true}, 
+                sanitised_at: {_is_null: true},
+                deleted_at: {_is_null: true}
+              }
+            ) {
+              id
+              data
+            }
+          }
+        }
+      }`
+    );
+
+    if (errors || !data) throw new Error(formatJSON({ data, errors }));
+    return data.temp_data_migrations_audit[0];
+  }
+
+  async updateQueuedFlow(flowId, liveFlowData, logs) {
+    const { data, errors } = await this.graphQL(
+      `mutation UpdateFlow ($flowId: uuid!, $liveData: jsonb!, $logs: String) {
+        update_flows_by_pk(pk_columns: {id: $flowId}, _set: {data: $liveData}) {
+          id
+        }
+        update_temp_data_migrations_audit_by_pk(pk_columns: {flow_id: $flowId}, _set: {updated: true, logs: $logs}) {
+          flow_id
+        }
+      }`,
+      {
+        flowId: flowId,
+        liveData: liveFlowData,
+        logs: logs,
+      }
+    );
+    if (errors || !data) throw new Error(formatJSON({ data, errors }));
+    return { ...data };
+  }
+
+  async updateQueuedPublishedFlow(flowId, liveFlowData, publishedFlowId, publishedFlowData, logs) {
+    const { data, errors } = await this.graphQL(
+      `mutation UpdateFlowAndPublishedFlow ($flowId: uuid!, $liveData: jsonb!, $publishedFlowId: Int!, $publishedData: jsonb!, $logs: String) {
+        update_flows_by_pk(pk_columns: {id: $flowId}, _set: {data: $liveData}) {
+          id
+        }
+        update_published_flows_by_pk(pk_columns: {id: $publishedFlowId}, _set: {data: $publishedData}) {
+          id
+        }
+        update_temp_data_migrations_audit_by_pk(pk_columns: {flow_id: $flowId}, _set: {updated: true, logs: $logs}) {
+          flow_id
+        }
+      }`,
+      {
+        flowId: flowId,
+        liveData: liveFlowData,
+        publishedFlowId: publishedFlowId,
+        publishedData: publishedFlowData,
+        logs: logs
+      }
+    );
+    if (errors || !data) throw new Error(formatJSON({ data, errors }));
+    return { ...data };
+  }
+
+  async updateSessionData(sessionId, sessionData) {
+    const { data, errors } = await this.graphQL(
+      `mutation UpdateLowcalSession ($id: uuid!, $data: jsonb!) {
+        update_lowcal_sessions_by_pk(pk_columns: {id: $id}, _set: {data: $data}) {
+          id
+        }
+      }`,
+      {
+        id: sessionId,
+        data: sessionData,
+      }
+    );
+    if (errors || !data) throw new Error(formatJSON({ data, errors }));
+    return {...data};
+  }
```

Comment:
`updateSessionData` is _not_ currently additionally updating the flows audit table, does that seem okay? We'll be able to check if a session was updated at least two ways: my crontab logs and the `updated_at` timestamp on the lowcal_sessions table

Reply:
I see no issues here 👍